### ISSUE TITLE
msktutil does not compile without udns on FreeBSD 12-STABLE

### ISSUE THREAD

**Issue by michael-o** [msktutil/msktutil#162]
I see:
```
msktname.cpp:187:20: error: use of undeclared identifier 'inet_ntop'
            (void) inet_ntop(ai->ai_family, ai->ai_addr, addrstr, INET6_ADDRSTRLEN);
                   ^
msktname.cpp:207:20: error: use of undeclared identifier 'inet_ntop'
            (void) inet_ntop(ai->ai_family, ai->ai_addr, addrstr, INET6_ADDRSTRLEN);
                   ^
msktname.cpp:229:45: warning: unused parameter 'domain' [-Wunused-parameter]
DnsSrvQuery::DnsSrvQuery(const std::string& domain, const std::string& service, const st...
                                            ^
msktname.cpp:229:72: warning: unused parameter 'service' [-Wunused-parameter]
DnsSrvQuery::DnsSrvQuery(const std::string& domain, const std::string& service, const st...
                                                                       ^
msktname.cpp:229:100: warning: unused parameter 'protocol' [-Wunused-parameter]
  ...std::string& domain, const std::string& service, const std::string& protocol)
```

with
```
# cc --version
FreeBSD clang version 10.0.1 (git@github.com:llvm/llvm-project.git llvmorg-10.0.1-0-gef32c611aa2)
Target: x86_64-unknown-freebsd12.2
Thread model: posix
InstalledDir: /usr/bin
```

I assume that some `#include` is missing.

**Comment by michael-o** [comment]
A possible fix:
```
# git diff
diff --git a/msktutil.h b/msktutil.h
index a0cd566..6e930ce 100644
--- a/msktutil.h
+++ b/msktutil.h
@@ -47,8 +47,10 @@
 #include <time.h>
 #include <limits.h>
 #include <netdb.h>
-#include <netinet/in.h>
+#include <sys/types.h>
 #include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
 #include <sys/utsname.h>
 #include <ldap.h>
 #include <list>
```

* RHEL 7 requires
```
#include <arpa/inet.h>`
```

* HP-UX 11.31 requires
```
#include <sys/socket.h>
#include <arpa/inet.h>
```

Do we need ifdefs here?

**Comment by oflebbe** [comment]
We would need ifdefs, but they should be checked by autoconf AC_CHECK_HEADERS , IMO.

As an followup question. @michael-o Do you know if we compile FreeBSD with github action automatically?

**Comment by michael-o** [comment]
No, we don't. We'd need to add Cirrus CI, this supports FreeBSD with Clang.

**Comment by oflebbe** [comment]
Indeed, Cirrus offers macos and FreeBSD. Would you like to try add it as an additional CI for compile smokes ?

**Comment by michael-o** [comment]
I will take a look next week into it!

**Comment by mjtrangoni** [comment]
@michael-o @oflebbe JFTR, GitHub Actions can also emulate it on MacOS, see [here](https://github.com/marketplace/actions/freebsd-vm)
It supports other architectures over QEMU, see [here](https://github.com/marketplace/actions/run-on-architecture)

Besides of that, you can run all distros on x86 with containers images.

**Comment by michael-o** [comment]
@mjtrangoni That's awesome! Since you know much better than I do, do you think you could provide a PR for this?

**Comment by mjtrangoni** [comment]
@michael-o yes sure, I will send you a PR.

**Comment by oflebbe** [comment]
Fixed by commit c691d88934598e2b9a3023ccc46efb2ab1ff10af
